### PR TITLE
Improve accept header handling for unknown types

### PR DESCRIPTION
**Pull request by ashokdelphia**
In the project where I'm using this, I particularly care about the case where the request prefers other types but has a lower quality match for JSON, e.g.:

    Accept: application/x-unknown, application/json; q=0.1

At the moment, it resolves using the first entry in the sorted list of types, even if there are better matches later on.

Sorry I failed to spot this when reviewing the q-value changes for #216.

### CODE REVIEW

File: health_check/views.py
Comment (codingjoe) [inline health_check/views.py]: How can that ever happen if the default is `*/*` see a couple lines above?
The only way this would happen, is if someone sends an empty accept header. In this case it might probably sense to respond with a 400.

File: tests/test_views.py
Comment (codingjoe) [inline tests/test_views.py:143]: The more the merrier, but I believe we have that already covered, don't we?

File: tests/test_views.py
Comment (codingjoe) [inline tests/test_views.py:177]: Same here, I believe this is covered, isn't it?

**Review by codingjoe — COMMENTED** [review]
Hi @ashokdelphia good to see that you are still following up on this issue. I did leave a couple of comments. Let me know what you think.

File: health_check/views.py
Comment (ashokdelphia) [reply health_check/views.py]: The Accept header can be absent, or could just contain types that we don't care about in the conditions above.

There's no requirement, as far as I remember, that the caller includes `*/*` in their list, so I think it's legitimate to fall off the bottom of the loop.

File: tests/test_views.py
Comment (ashokdelphia) [reply tests/test_views.py:177]: This test was failing with the old code, because the loop was only considering the first element in the sorted list.

File: tests/test_views.py
Comment (ashokdelphia) [reply tests/test_views.py:143]: It is, but I don't think this particular case is quite there in the other tests.

File: health_check/views.py
Comment (codingjoe) [reply health_check/views.py]: I would honestly like to see a test that breaks it and I still believe in this case we should respond with a 400, since we can't serve a response that would satisfy the request.

Comment (ashokdelphia) [reply health_check/views.py]: Apologies. I thought I replied to this yesterday.

The old code did return HTML if the 'top' result from the request's accept list didn't match the explicit lists, so my change here was trying to keep that behaviour, but keep looking at the rest of the acceptable types.

I think you're right, though: it's reasonable to make that case a 400-series response. I think a 406 would be appropriate. I'll add a covering test and change the behaviour.

Comment (codingjoe) [reply health_check/views.py]: @ashokdelphia, OK, go for it!

Comment (ashokdelphia) [reply health_check/views.py]: Sorry for the delay responding.

I pushed up the changes.

I'm a little concerned this would be a breaking change for anyone currently hitting the endpoint with an unrecognized Accept header, but if you think that's okay, then feel free to merge.

File: health_check/views.py
Comment (codingjoe) [inline health_check/views.py]:
```suggestion
        return HttpResponse('Not Acceptable: Supported content types: plain/html, application/json', status=http.HTTPStatus.NOT_ACCEPTABLE, content_type='text/plain')
```

**Review by codingjoe — APPROVED** [review]
Cool, feature wise, this looks perfect. I just made some suggestions regarding the text.
Let me know once you are happy and I will merge and release it.

File: health_check/views.py
Comment (ashokdelphia) [reply health_check/views.py]: Thanks for the suggestion. I applied it with a minor tweak.

**Comment by ashokdelphia** [comment]
@codingjoe I think that's good to go now, if you're happy with it.

**Comment by codingjoe** [comment]
@ashokdelphia I'll merge this once you fix the test :)

**Comment by ashokdelphia** [comment]
Apologies for missing the lint failure. That should be fine now.

**Comment by codingjoe** [comment]
Released in [3.10.2](https://github.com/KristianOellegaard/django-health-check/releases/tag/3.10.2)

**Pull request merged**
